### PR TITLE
Added Date Component

### PR DESCRIPTION
**Pull request by kalismeras61**
added date component need to add intl package for formatting date. In xml looks like in below.
       ` <date id="formDate" label="Form Date" dateValue="" firstDate="2010" lastDate="2030" format="yyyy-dd-MM"></date>`

### CODE REVIEW

**Comment by OndrejKunc** [comment]
Thanks for the merge request, the component looks great! :) I actually wanted to create such component myself but didn't have time for it yet.

I noticed just two small things:

1) In the components package, I use the yaml files to generate all the models using https://github.com/OndrejKunc/flutter_dynamic_forms/tree/master/packages/dynamic_forms_generator

This means, that I add the YAML file for each component and call `flutter packages pub run build_runner build`. This will generate both the model and the parser for the component. That's why there is a .g in the file name of the component, so it is clear that it is generated.

So if you don't want to add the YAML file, then please remove the .g from the file name and the `// GENERATED CODE - DO NOT MODIFY BY HAND` comment.

If you, however, want to add the YAML file, just be aware that there is currently a bug that happens when you call the build command which causes an example project to show multiple errors and the only solution is to delete .dart_tool folder.

2) We also support type `DateTime` in our expressions, so instead of using `String` for the dateValue or `int` for the startDate and endDate I would prefer using DateTime for all of them. You would parse it in the parser class to the dateTime using DateTime.parse or tryParse. There is even support for dateTime in generators, see https://github.com/OndrejKunc/flutter_dynamic_forms/blob/master/packages/dynamic_forms_generator/example/lib/check_box/check_box.yaml but I am afraid it doesn't support `default DateTime.now()` so you probably don't want to use the yaml file until this is sorted.

The reason for this is that it would be nice to define startDate and endDate by exact date, not just by the year and also if the properties are of type DateTime then you can use them in expressions, because we support all sort of operators on the dateTime in our expression language, see this old merge request: https://github.com/OndrejKunc/flutter_dynamic_forms/pull/9


If this seems like too much work then I think I can merge it anyway if you just address the issues in point one by renaming files and removing comments. I can convert it to the DateTime myself, but it is up to you :)

Thanks again.

**Comment by OndrejKunc** [comment]
FYI: I just came up with an example of what you can do if all the properties are of type DateTime:
```xml
<date id="formDate1" label="Form Date" firstDate="2010-02-27" lastDate="2030-02-27" format="yyyy-dd-MM"/>
<date id="formDate2" label="Form Date 2" format="yyyy-dd-MM">
    <date.firstDate>
        <expression>@formDate1.dateValue - duration("P5D")</expression>
    </date.firstDate>
    <date.lastDate>
        <expression>@formDate1.dateValue + duration("P5D")</expression>
    </date.lastDate>
</date>
```
This means you would be able to set the firstDate in the formDate2 based on the selected dateValue in formDate1 to the 5 days before this date and the lastDate 5 days after dateValue. All on the client side :)....

**Comment by kalismeras61** [comment]
Hi @OndrejKunc yes i know when automatically build it is generated g. dar files. I just wanted the same names with others components but will change it. :)

About DateTime yes you are right I didn't think of about expressions so why i just used String but after your explains i have to change it with DateTime.

Thanks.

**Comment by kalismeras61** [comment]
Hi @OndrejKunc you can check new pr i change value type with DateTime and also change dateValue propertyname with just value and one more thing i added initialDate propertyNameç

**Comment by OndrejKunc** [comment]
Hi @kalismeras61 ,
Are you sure you pushed those changes? Only changes in last two commits I can see are changes in the example xml....

**Comment by kalismeras61** [comment]
> Hi @kalismeras61 ,
> Are you sure you pushed those changes? Only changes in last two commits I can see are changes in the example xml....

Yep , click the commits tab as you see i have 5 commits 2 one is created today.

**Comment by OndrejKunc** [comment]
No problem :)

**Comment by kalismeras61** [comment]
@OndrejKunc now i think everything looks good :) Please check the PR.

File: packages/flutter_dynamic_forms_components/lib/src/components/date/date_parser.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      initialDate: parserNode.getValue(
+        "initialDate",
+        (date) => DateTime.parse(date),
+        () => DateTime.now(),
+        isImmutable: true,
+      ),
+      firstDate: parserNode.getValue(
+          "firstDate",
+          (date) => DateTime.parse(date),
+          () => DateTime.parse("1969-07-20 20:18:04Z"),
```
Comment (OndrejKunc) [inline packages/flutter_dynamic_forms_components/lib/src/components/date/date_parser.dart]: I think there is no need to call DateTime.parse and DateTime can be created directly, same applies to the lastDate

File: packages/flutter_dynamic_forms_components/lib/src/components/date/reactive_date_renderer.dart [hunk 2]
```diff
@@ -33,20 +32,20 @@ class ReactiveDateRenderer extends FormElementRenderer<model.Date> {
                 final DateTime picked = await showDatePicker(
                   context: context,
                   firstDate: element.firstDate != null
-                      ? DateTime(element.firstDate)
-                      : DateTime(1900),
+                      ? element.firstDate
+                      : DateTime(1979 - 01 - 01),
```
Comment (OndrejKunc) [inline packages/flutter_dynamic_forms_components/lib/src/components/date/reactive_date_renderer.dart]: You probably meant to put comma there:  `DateTime(1979, 1, 1)`
Same with last date.

File: packages/flutter_dynamic_forms_components/lib/src/components/date/date_parser.dart (hunk 1 above)
Comment (kalismeras61) [reply packages/flutter_dynamic_forms_components/lib/src/components/date/date_parser.dart]: Which one default value ? converter ? If you mean default value the time is int type so we need to parse it. ? am i wrong?

File: packages/flutter_dynamic_forms_components/lib/src/components/date/reactive_date_renderer.dart (hunk 2 above)
Comment (kalismeras61) [reply packages/flutter_dynamic_forms_components/lib/src/components/date/reactive_date_renderer.dart]: yes you are right. i changed it.

File: packages/flutter_dynamic_forms_components/lib/src/components/date/date_parser.dart (hunk 1 above)
Comment (kalismeras61) [reply packages/flutter_dynamic_forms_components/lib/src/components/date/date_parser.dart]: Do you mean like this?
```
 firstDate: parserNode.getValue(
        "firstDate",
        (date) => DateTime.parse(date),
        () => DateTime(1969, 07, 20),
        isImmutable: true,
      ),
      lastDate: parserNode.getValue(
        "lastDate",
        (date) => DateTime.parse(date),
        () => DateTime(2050, 07, 20),
        isImmutable: true,
      ),
```

Comment (OndrejKunc) [reply packages/flutter_dynamic_forms_components/lib/src/components/date/date_parser.dart]: yes, that's what I meant

**Comment by kalismeras61** [comment]
Finally :)  Thanks again for this amazing library!

**Comment by OndrejKunc** [comment]
I thank you for the contribution, it's nice to see that someone else except our team finds this package useful :)

FYI: I already published it on the pub

**Comment by kalismeras61** [comment]
Your welcome. I think i will work on it about more components :)
I am planning date range picker and also searchable dropdown component as soon.

**Pull request merged**